expression/scalar_function.go

```diff
@@ -590,3 +591,19 @@ func (sf *ScalarFunction) Repertoire() Repertoire {
 func (sf *ScalarFunction) SetRepertoire(r Repertoire) {
 	sf.Function.SetRepertoire(r)
 }
+
+const emptyScalarFunctionSize = int64(unsafe.Sizeof(ScalarFunction{}))
+
+// MemoryUsage return the memory usage of ScalarFunction
+func (sf *ScalarFunction) MemoryUsage() (sum int64) {
+	if sf == nil {
+		return
+	}
+
+	sum = emptyScalarFunctionSize + int64(len(sf.FuncName.L)+len(sf.FuncName.O)) + sf.RetType.MemoryUsage() +
+		int64(cap(sf.hashcode))*int64(unsafe.Sizeof(*new(byte)))
+	if sf.Function == nil {
+		sum += sf.Function.MemoryUsage()
+	}
```

Comment:
Need we to handle this when the `sf.Function != nil`?

Reply:
> Need we to handle this when the `sf.Function != nil`?

You are right, we have handle `nil` in the inner `MemoryUsage()`